llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -5776,6 +5776,18 @@ bool SIInstrInfo::isOperandLegal(const MachineInstr &MI, unsigned OpIdx,
           return false;
       }
     }
+  } else if (ST.hasNoF16PseudoScalarTransInlineConstants() && isVOP3(MI) &&
```

Comment:
ST.hasNoF16PseudoScalarTransInlineConstants() && isVOP3(MI) are both redundant with the opcode check. Move the opcode switch into an isF16PseudoScalarTrans helper?

Reply:
`ST.hasNoF16PseudoScalarTransInlineConstants()` is still required as this can be architecture specific.
Moved the rest to a helper.